client/indexer-db/src/models/file.rs

```diff
@@ -183,10 +186,26 @@ impl File {
         file_key: impl AsRef<[u8]>,
     ) -> Result<(), diesel::result::Error> {
         let file_key = file_key.as_ref().to_vec();
+
+        // Get file info before deletion
+        let file_info: Option<(i64, i64)> = file::table
+            .filter(file::file_key.eq(&file_key))
+            .select((file::bucket_id, file::size))
+            .first(conn)
+            .await
+            .optional()?;
+
+        // Delete the file
         diesel::delete(file::table)
             .filter(file::file_key.eq(file_key))
             .execute(conn)
             .await?;
+
+        // Update bucket counts if file was found
+        if let Some((bucket_id, file_size)) = file_info {
+            Bucket::decrement_file_count_and_size(conn, bucket_id, file_size).await?;
+        }
+
```

Comment:
Out of curiosity, is this happening atomically?

Reply:
from `handle_finality_notification` -> `index_block` which then trickles the data to everything else https://github.com/Moonsong-Labs/storage-hub/blob/a651b502cf767e00f695ce79daf9b67cde842dc7/client/indexer-service/src/handler.rs#L121-L134